src/devices/sound/c140.cpp

```diff
@@ -158,12 +162,62 @@ void c140_device::device_start()
 	}
 }
 
+void c219_device::device_start()
+{
+	c140_device::device_start();
+	std::fill(std::begin(m_c219bank), std::end(m_c219bank), 0);
 
-//-------------------------------------------------
-//  sound_stream_update - handle a stream update
-//-------------------------------------------------
+	save_item(NAME(m_c219bank));
+}
 
-void c140_device::sound_stream_update(sound_stream &stream, stream_sample_t **inputs, stream_sample_t **outputs, int samples)
+void c140_device::device_reset()
+{
+	std::fill(std::begin(m_REG), std::end(m_REG),0);
+
+	for(int i = 0; i < MAX_VOICE; i++)
+	{
+		init_voice(&m_voi[i]);
+	}
+}
+
+void c219_device::device_reset()
+{
+	c140_device::device_reset();
+	std::fill(std::begin(m_c219bank), std::end(m_c219bank), 0);
+}
+
+void c140_device::device_post_load()
+{
+	device_clock_changed();
+}
+
+void c140_device::device_clock_changed()
+{
+	int old_rate = m_sample_rate;
+	m_sample_rate = m_baserate = clock();
+	if (old_rate < m_sample_rate)
+	{
+		std::fill(m_mixer_buffer_left.begin(), m_mixer_buffer_left.end(), 0);
+		std::fill(m_mixer_buffer_right.begin(), m_mixer_buffer_right.end(), 0);
+		m_mixer_buffer_left.resize(m_sample_rate);
+		m_mixer_buffer_right.resize(m_sample_rate);
+	}
+	if (m_stream != nullptr)
+		m_stream->set_sample_rate(m_sample_rate);
+	else
+		m_stream = machine().sound().stream_alloc(*this, 0, 2, m_sample_rate);
+		
+	if (old_rate > m_sample_rate)
+	{
+		std::fill(m_mixer_buffer_left.begin(), m_mixer_buffer_left.end(), 0);
+		std::fill(m_mixer_buffer_right.begin(), m_mixer_buffer_right.end(), 0);
```

Comment:
Why are you filling the buffer before the resize here?  The new size will be smaller, so you're doing additional work for nothing.  You'd save (a small amount of) cycles by doing it after the resize.